src/io/iohk/scalanet/test/peergroup/TLSPeerGroupSpec.scala

```diff
@@ -0,0 +1,76 @@
+package io.iohk.scalanet.peergroup
+
+import io.iohk.decco.auto._
+import io.iohk.scalanet.NetUtils._
+import io.iohk.scalanet.TaskValues._
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.Matchers._
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.concurrent.ScalaFutures._
+import org.scalatest.{BeforeAndAfterAll, FlatSpec}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.Random
+
+class TLSPeerGroupSpec extends FlatSpec with BeforeAndAfterAll {
+
+  implicit val patienceConfig: ScalaFutures.PatienceConfig = PatienceConfig(5 seconds)
+
+  behavior of "TCPPeerGroup"
+
+  it should "send and receive a message when client auth is disabled / false" in
+    withTwoRandomTLSPeerGroups[String](false) { (alice, bob) =>
+      println(s"Alice is ${alice.processAddress}, bob is ${bob.processAddress}")
+      val alicesMessage = Random.alphanumeric.take(1024).mkString
+      val bobsMessage = Random.alphanumeric.take(1024).mkString
+
+      bob.server().foreachL(channel => channel.sendMessage(bobsMessage).evaluated).runAsync
+      val bobReceived: Future[String] = bob.server().mergeMap(channel => channel.in).headL.runAsync
+      val aliceClient = alice.client(bob.processAddress).evaluated
+      Thread.sleep(2000)
+      val aliceReceived = aliceClient.in.headL.runAsync
+      aliceClient.sendMessage(alicesMessage).evaluated
+
+      bobReceived.futureValue shouldBe alicesMessage
+      aliceReceived.futureValue shouldBe bobsMessage
+    }
+
+  it should "send and receive a message when client auth is enabled or true" in
```

Comment:
See comment above about refactoring into a single test with two configs, one for authenticated one for unauthenticated.